crates/starknet-devnet/tests/common/background_devnet.rs

```diff
@@ -126,17 +130,16 @@ impl BackgroundDevnet {
                 .spawn()
                 .expect("Could not start background devnet");
 
-        let healthcheck_uri =
-            format!("{}{HEALTHCHECK_PATH}", devnet_url.as_str()).as_str().parse::<Uri>()?;
+        let healthcheck_uri = format!("{}{HEALTHCHECK_PATH}", devnet_url.as_str()).to_string();
+        let reqwest_client = Client::new();
 
-        let http_client = Client::new();
         let max_retries = 30;
         for _ in 0..max_retries {
-            if let Ok(alive_resp) = http_client.get(healthcheck_uri.clone()).await {
+            if let Ok(alive_resp) = reqwest_client.get(&healthcheck_uri).send().await {
                 assert_eq!(alive_resp.status(), StatusCode::OK);
                 println!("Spawned background devnet at port {free_port}");
                 return Ok(BackgroundDevnet {
-                    http_client,
+                    reqwest_client: ReqwestClient::new(devnet_url.clone(), reqwest_client),
```

Comment:
Why can't we reuse the client created in L134, but have to create a new one in L142?

Reply:
it is reused

Reply:
It is reused by this `ReqwestClient` wrapper, but why do we even need the wrapper? And why is it called `ReqwestClient` when there already is `reqwest::Client`?

Reply:
take a look at ReqwestClient functionality

Reply:
I sense it is because of the sync/async stuff; not familiar with this:
```rust
#[async_trait::async_trait]
```

Reply:
90% of the time a post or get request is sent there is a response. ReqwestClient is a wrapper that returns the response or an error, without manually deserialising on the caller side